apps/prairielearn/src/ee/lib/aiQuestionGeneration.ts

```diff
@@ -0,0 +1,90 @@
+import { OpenAI } from 'openai';
+
+import { loadSqlEquiv, queryRows } from '@prairielearn/postgres';
+
+import { QuestionGenerationContextEmbeddingSchema } from '../../lib/db-types.js';
+import { createServerJob } from '../../lib/server-jobs.js';
+
+import { createEmbedding, openAiUserFromLocals, vectorToString } from './contextEmbeddings.js';
+
+const sql = loadSqlEquiv(import.meta.url);
+
+export async function generateQuestion(
```

Comment:
Also add JSDoc comments for functions in these files.

Reply:
Done in ff0b8fa.